### PR TITLE
Remove unused boost includes

### PR DESCRIPTION
**Pull request by agutenkunst**
## Description

Removes boost includes that are not used.

### Things to add, update or check by the maintainers before this PR can be merged.

* [x] ~Public api function documentation~
* [x] ~Architecture documentation reflects actual code structure~
* [x] ~[Tutorials](https://wiki.ros.org/pilz_robots/Tutorials/)~
* [x] ~Overview on [ROS wiki](https://wiki.ros.org/pilz_robots)~
* [x] ~Package Readme ([example pilz_robots](https://github.com/PilzDE/pilz_robots/blob/melodic-devel/README.md))~
* [x] Good commit messages ([some tips](https://dev.to/jacobherrington/how-to-write-useful-commit-messages-my-commit-message-template-20n9))
* [x] ~CHANGELOG.rst updated~
* [x] ~Copyright headers~
* [x] ~Examples~

### Review Checklist
* [x] ~Soft- and hardware architecture (diagrams and description)~
* [x] ~Test review (test plan and individual test cases)~
* [x] ~Documentation describes purpose of file(s) and responsibilities~
* [x] Code (coding rules, style guide)

### Release planning (please answer)
* [x] ~When is the new feature released?~
* [x] ~Which dependent packages do have to be released simultaneously?~

### CODE REVIEW

File: test/hw_tests/hwtest_scan_compare.cpp [hunk 1]
```diff
@@ -17,7 +17,6 @@
 #include <gtest/gtest.h>
 
 #include <boost/filesystem.hpp>
-#include <boost/function.hpp>
 #include <boost/bind.hpp>
```
Comment (jschleicher) [inline test/hw_tests/hwtest_scan_compare.cpp:20]:
```suggestion
```
Where is bind used in `scan_compare.cpp`? Are there more unused boost modules?

File: standalone/include/psen_scan_v2_standalone/udp_client.h [hunk 2]
```diff
@@ -33,7 +33,6 @@
 #endif
 
 #include <boost/asio.hpp>
-#include <boost/asio/high_resolution_timer.hpp>
 #include <boost/bind.hpp>
```
Comment (jschleicher) [inline standalone/include/psen_scan_v2_standalone/udp_client.h:36]:
```suggestion
```

Maybe also replace `boost::bind` by `std::bind`? (See https://stackoverflow.com/questions/8924149/should-stdbind-be-compatible-with-boostasio)

File: test/hw_tests/hwtest_scan_compare.cpp (hunk 1 above)
Comment (agutenkunst) [reply test/hw_tests/hwtest_scan_compare.cpp]: Not used. Removed.

File: standalone/include/psen_scan_v2_standalone/udp_client.h (hunk 2 above)
Comment (agutenkunst) [reply standalone/include/psen_scan_v2_standalone/udp_client.h:36]: Surely could be done. But I try to keep it simple by starting in small steps and removing the obvious ones first.

**Comment by agutenkunst** [comment]
Please review und check the checkboxes in the PR description!

**Review by cdoehn — APPROVED** [review]

**Pull request merged**